### DISCUSSION TITLE
How to customise matplotlib / iris.qplot label?

### DISCUSSION THREAD

**Discussion by maczik** [SciTools/iris#4748] [Q&A] [answered]
I am trying to customise the label of my figure so it spans the same range regardless of the data showcased. Uniform labelling is important as I am making an animation over time of my simulation results.

For the purposes of this post I am using the sample data. Here is my code:

> 
> import matplotlib.pyplot as plt
> import iris
> import iris.quickplot as qplt
> 
> fname = iris.sample_data_path("air_temp.pp")
> temperature_cube = iris.load_cube(fname)
> 
> ##Draw the contour with 25 levels.
> qplt.contourf(temperature_cube, 25)
> 
> ##Add coastlines to the map created by contourf.
> plt.gca().coastlines()
> 
> plt.show()

And here is the result: 
<img width="367" alt="kelvin" src="https://user-images.githubusercontent.com/100720430/169069788-b97d4442-4fdd-4de1-a560-4719d435da34.png">

I want to manually set the label range of the colour bar under the map, lets say from 230 to 310. I have not had any luck figuring this one out yet, any help would be very appreciated.

Thank you for considering the problem.

**Comment by rcomer** [comment, marked as answer]
Hi @maczik, the Iris plotting functions are wrappers to the matplotlib plotting functions.  So to find all the configuration options you can look at the [Matplotlib docs](https://matplotlib.org/stable/api/_as_gen/matplotlib.pyplot.contourf.html?highlight=contourf#matplotlib.pyplot.contourf).  In this case I think you want to set the `levels` keyword, and/or `vmin` and `vmax`.

The `iplt.contourf` call in this Gallery example seems similar to what you want:
https://scitools-iris.readthedocs.io/en/stable/generated/gallery/meteorology/plot_lagged_ensemble.html#sphx-glr-generated-gallery-meteorology-plot-lagged-ensemble-py

**Comment by pp-mo** [comment]
Looks answered.  Archiving this
